### ISSUE TITLE
Layer freezing documentation/example

### ISSUE THREAD

**Issue by Maratyszcza** [NervanaSystems/neon#344]
Release notes for **Neon v1.1.0 (2015-10-30)** mention **Support for freezing layer stacks**.
I couldn't find any documentation or examples on using this feature. Is it still supported? Which method should I call to freeze a layer?

**Comment by hanlin-nervana** [comment]
To freeze a layer, we typically using Multiple Optimizers (e.g. see https://github.com/NervanaSystems/neon/blob/master/examples/faster-rcnn/train.py#L101) to set the learning rate of particular layers to 0.

**Comment by Maratyszcza** [comment]
Thanks for the pointer. Is Neon smart enough to avoid allocation memory for gradients and reuse deltas when frozen layers are in the beginning of the network?

**Comment by hanlin-nervana** [comment]
No layers that are frozen still undergo the same bprop computations, just the weights are not updated. If this is absolutely necessary, you could consider having a `frozen` attribute and writing a new container that controls the [allocation](https://github.com/NervanaSystems/neon/blob/master/neon/layers/container.py#L297) and [bprop](https://github.com/NervanaSystems/neon/blob/master/neon/layers/container.py#L339) methods appropriately to achieve this optimization.

Is your use case where you have a large front-end model that is frozen (e.g. VGG) and would like to reduce its memory consumption for the follow-on components?

**Comment by Maratyszcza** [comment]
Yes, thats exactly my use-case.

**Comment by hanlin-nervana** [comment]
Thanks. Unfortunately we do not have anything built in for that situation, but we certainly welcome contributions. When training on large datasets across many epochs where the feature-extractor (e.g. VGG) is fixed, you may also consider just generating a new dataset with the features pre-computed to eliminate the fprop compute time and the memory footprint of the front-end. For example, see the YouTube-8M dataset.
